strawberry/union.py

```diff
@@ -171,9 +171,12 @@ def _resolve_union_type(
         return _resolve_union_type
 
 
+Types = TypeVar("Types", bound=Type)
+
+
 def union(
-    name: str, types: Tuple[Type, ...], *, description: str = None
-) -> StrawberryUnion:
+    name: str, types: Tuple[Types, ...], *, description: str = None
+) -> Union[Types]:  # type: ignore
```

Comment:
Add a comment explaining that the typing of `types` is perfect right now, but will change in the future with [Variadic Generics](https://www.python.org/dev/peps/pep-0646/).